examples/site/log_stackdriver/log_stackdriver.cc

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_log_stackdriver]
+#include <google/cloud/functions/cloud_event.h>
+#include <cppcodec/base64_rfc4648.hpp>
+#include <nlohmann/json.hpp>
+#include <iostream>
+
+namespace gcf = ::google::cloud::functions;
+
+void log_stackdriver(gcf::CloudEvent event) {  // NOLINT
+  if (event.data_content_type().value_or("") != "application/json") {
+    std::cerr << "expected application/json data";
+    return;
+  }
+  auto const payload = nlohmann::json::parse(event.data().value_or("{}"));
+  auto const data = cppcodec::base64_rfc4648::decode<std::string>(
+      payload["message"]["data"].get<std::string>());
+  auto const log_entry = nlohmann::json::parse(data);
+  std::cout << "Method: " << log_entry.value("methodName", "")
+            << "\nResource: " << log_entry.value("resourceName", "")
+            << "\nInitiation: "
```

Comment:
Q. "Initiation"?  Maybe "AuthenticationPrincipal"??

Reply:
I stole this from the python example:

https://cloud.google.com/functions/docs/calling/logging#functions_log_stackdriver-python

which lucky for me matches node:

https://cloud.google.com/functions/docs/calling/logging#functions_log_stackdriver-nodejs

but I do not think matches the other two languages :disappointed: 

https://cloud.google.com/functions/docs/calling/logging#functions_log_stackdriver-go

I am going to go with a generic print because it seems simpler and consistent with at least some examples.